### PR TITLE
Search by address implemented

### PR DESCRIPTION
**Pull request by sehoven**
User can search requests based on an address.

Solves: [#141](https://github.com/CMPUT301F16T01/Carrier/issues/141)

### CODE REVIEW

File: app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java [hunk 1]
```diff
@@ -200,12 +202,30 @@ protected RequestList doInBackground(Location... search_parameters) {
...
+            return addressList;
+        }
+    }
+
     /**
-     * Get all of a rider's requests including a filter by status
-=======
      * Get all of a rider's requests filtered by status
      * @see RequestController#fetchRequestsWhereRider(User, Integer...)
->>>>>>> 92fda80c294f949b664f49d206e3dedeea1f17a7
```
Comment (Hreherch) [inline app/src/main/java/comcmput301f16t01/github/carrier/Requests/ElasticRequestController.java:208]: I noticed you caught this here but I commented in your other pull request about it.

File: app/src/main/java/comcmput301f16t01/github/carrier/Searching/SearchAddressChoiceActivity.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+import comcmput301f16t01.github.carrier.CarrierLocation;
+import comcmput301f16t01.github.carrier.R;
+import comcmput301f16t01.github.carrier.Requests.DriverViewRequestActivity;
+import comcmput301f16t01.github.carrier.Requests.Request;
+import comcmput301f16t01.github.carrier.Requests.RequestController;
+
+public class SearchAddressChoiceActivity extends AppCompatActivity {
+    final Activity activity = SearchAddressChoiceActivity.this;
+    // TODO http://stackoverflow.com/questions/32444863/google-gson-linkedtreemap-class-cast-to-myclass
```
Comment (Hreherch) [inline app/src/main/java/comcmput301f16t01/github/carrier/Searching/SearchAddressChoiceActivity.java]: Reminder for your TODO

File: app/src/main/java/comcmput301f16t01/github/carrier/Searching/SearchAddressChoiceActivity.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import comcmput301f16t01.github.carrier.CarrierLocation;
+import comcmput301f16t01.github.carrier.R;
+import comcmput301f16t01.github.carrier.Requests.DriverViewRequestActivity;
+import comcmput301f16t01.github.carrier.Requests.Request;
+import comcmput301f16t01.github.carrier.Requests.RequestController;
+
+public class SearchAddressChoiceActivity extends AppCompatActivity {
```
Comment (Hreherch) [inline app/src/main/java/comcmput301f16t01/github/carrier/Searching/SearchAddressChoiceActivity.java:33]: Needs top-level JavaDoc?

File: app/src/main/java/comcmput301f16t01/github/carrier/Searching/SearchActivity.java [hunk 4]
```diff
@@ -58,16 +69,93 @@ public void onClick(DialogInterface dialogInterface, int i) {
...
         startActivity(intent);
     }
+
+    /**
+     * This will open a dialog to search for an address. The selected address
+     * will be shown on a map to confirm the location. This will create a
+     * query for available requests.
+     * @param view
+     */
+    // TODO decide how to allow the user to enter the address
```
Comment (Hreherch) [inline app/src/main/java/comcmput301f16t01/github/carrier/Searching/SearchActivity.java]: I believe you answered this TODO.

**Review by Hreherch — CHANGES_REQUESTED** [review]
Looking good! 
Minor edits, mayhaps. 

One enhancement is that I rather be able to type in my search in the SearchAddressChoiceActivity. This way I can type in an address, hit go (this would be like a google search, where the results appear immediately below), and immediately see on the same activity the potential locations my address brought up.

**Comment by sehoven** [comment]
Just letting you know that I've seen your comments and I will address the major one first (search on same activity) sometime today.

File: app/src/main/java/comcmput301f16t01/github/carrier/Searching/SearchAddressChoiceActivity.java (hunk 3 above)
Comment (sehoven) [reply app/src/main/java/comcmput301f16t01/github/carrier/Searching/SearchAddressChoiceActivity.java:33]: Like a javadoc on the activity class?

**Review by Hreherch — APPROVED** [review]

**Pull request merged**